pkg/storage/gcp/gcp.go

```diff
@@ -3,40 +3,48 @@ package gcp
 import (
 	"context"
 	"fmt"
+	"net/http"
 	"net/url"
 
 	"cloud.google.com/go/storage"
 	"github.com/gomods/athens/pkg/config/env"
-	"google.golang.org/api/option"
+	"github.com/gomods/athens/pkg/errors"
+	"google.golang.org/api/googleapi"
 )
 
 // Storage implements the (./pkg/storage).Backend interface
 type Storage struct {
 	bucket       Bucket
 	baseURI      *url.URL
 	closeStorage func() error
+	projectID    string
 }
 
-// NewWithCredentials returns a new Storage instance authenticated using the provided
+// New returns a new Storage instance authenticated using the provided
 // ClientOptions. The bucket name to be used will be loaded from the
 // environment variable ATHENS_STORAGE_GCP_BUCKET.
 //
-// The ClientOptions should provide permissions sufficient to read, write and
-// delete objects in google cloud storage for your project.
-func NewWithCredentials(ctx context.Context, cred option.ClientOption) (*Storage, error) {
-	storage, err := storage.NewClient(ctx, cred)
+// If you're on your dev machine, make sure you're logged in to gcloud.
+// If you're outside GCP, make sure you have GOOGLE_APPLICATION_CREDENTIALS set.
+func New(ctx context.Context) (*Storage, error) {
+	const op errors.Op = "gcp.New"
+	storage, err := storage.NewClient(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("could not create new storage client: %s", err)
+		return nil, errors.E(op, fmt.Errorf("could not create new storage client: %s", err))
 	}
-	bucketname, err := env.GcpBucketName()
+	bucketname, err := env.GCPBucketName()
 	if err != nil {
-		return nil, err
+		return nil, errors.E(op, err)
 	}
 	u, err := url.Parse(fmt.Sprintf("https://storage.googleapis.com/%s", bucketname))
 	if err != nil {
-		return nil, err
+		return nil, errors.E(op, err)
 	}
 	bkt := gcpBucket{storage.Bucket(bucketname)}
+	err = bkt.Create(ctx, env.GCPPRojectID(), nil)
```

Comment:
What happens if the project ID is not set?

Reply:
@robjloranger it returns `googleapi: Error 400: Unknown project id: , invalid` which is clear enough for us to fatal out of the boot up with that message.